domain/network/service/space.go

```diff
@@ -32,6 +33,41 @@ func NewSpaceService(st State, logger Logger) *SpaceService {
 	}
 }
 
+// FilterHostPortsForManagementSpace filters the collection of API addresses
+// based on the configured management space for the controller.
+// If there is no space configured, or if one of the slices is filtered down
+// to zero elements, just use the unfiltered slice for safety - we do not
+// want to cut off communication to the controller based on erroneous config.
+func (s *SpaceService) FilterHostPortsForManagementSpace(
```

Comment:
This doesn't need a service method. The only use you make of the state is to look up a space.

The only place that you call it is the bootstrap worker, where you already have all the `SpaceInfos`.
You can just find the space by name there an use it to filter without going back to state via the service.